src/QirRuntime/lib/QIR/CMakeLists.txt

```diff
@@ -22,8 +22,10 @@ endif()
 # Step 1: create qir-rt-support lib from the C++ sources
 #
 set(rt_sup_source_files
+  "allocationsTracker.cpp"
   "arrays.cpp"
   "callables.cpp"
+  "context.cpp"

```

Comment:
Minor (feel free to ignore):
As I understand, here the alphabetic ordering of the files is used (files are compiled in the alphabetic order).
There is also another point of view about the compilation order, that you may find worth knowing. See [Build the Most Failing Files First](https://github.com/kuzminrobin/code_review_notes/blob/master/misc_posts.md#build-the-most-failing-files-first) section.

Reply:
Right now the code churns pretty much across the board so I feel it's too early to optimize the build system yet. Currently, the longest steps of the build are catch2 tests (because I'm using a single header include which is known to be sub-optimal as far as build-time is concerned -- this one will be worth looking into at some point) and compilation of *.ll files. For the latter I don't know how to speed things up but I also don't expect to grow the body of these tests much.